frontend/ContextMenu.js

```diff
@@ -58,16 +63,39 @@ class ContextMenu extends React.Component {
     this.props.hideContextMenu();
   }
 
+  getHeight = element => {
+    if (!element) {
+      return;
+    }
+
+    const elementHeight = element.querySelector('ul').clientHeight;
+    const windowHeight = window.innerHeight;
+
+    if (this.state.elementHeight === elementHeight && this.state.windowHeight === windowHeight) {
+      return;
+    }
+
+    this.setState({
+      elementHeight: elementHeight,
+      windowHeight: windowHeight,
+    });
+  };
+
   render() {
-    const {theme} = this.context;
-    const {items, open, pos} = this.props;
+    const { theme } = this.context;
+    const { items, open, pos } = this.props;
+    const { elementHeight, windowHeight } = this.state;
+
+    if (pos && (pos.y + elementHeight) > windowHeight) {
+      pos.y -= elementHeight;
+    }
 
     if (!open) {
       return <div style={styles.hidden} />;
     }
 
     return (
-      <div style={styles.backdrop} onClick={this.handleBackdropClick}>
+      <div style={styles.backdrop} onClick={this.handleBackdropClick} ref={this.getHeight}>
```

Comment:
nit: Not a fan of the name of this method. It isn't a getter; it updates state. It's also a bit weird to have a ref-setter named "get" something. My preference would be to call it something like `_setRef`